cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4MeasureReportBuilder.java

```diff
@@ -400,13 +401,20 @@ protected void buildStratifier(
         }
 
         if (!stratifierDef.components().isEmpty()) {
+            // StratifierComponentDef (expression name, id) -> Map<Subject String, Expression Result>
+
             // Component Stratifier
             // one or more criteria expression defined, one set of criteria results per component specified
             // results of component stratifier are an intersection of membership to both component result sets
-
-            List<Map<String, CriteriaResult>> subjectCompValues = stratifierDef.components().stream()
-                    .map(StratifierComponentDef::getResults)
-                    .toList();
+            List<Map<String, Pair<CriteriaResult, StratifierComponentDef>>> subjectCompValues =
```

Comment:
I'll be honest with you:  I'm having a really hard time following this new code.  While I'm pretty sure it does what you intend to do, it's hard to read and maintain.  I strongly suggest writing one or more classes dedicated this new functionality, wrapping the myriad of nested structures with an API with a clear set of contracts that can be easily unit tested.